monai/data/meta_tensor.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import warnings
+from typing import Callable
+
+import torch
+
+from monai.data.meta_obj import MetaObj, get_track_meta, get_track_transforms
+from monai.utils.enums import PostFix
+
+__all__ = ["MetaTensor"]
+
+
+class MetaTensor(MetaObj, torch.Tensor):
+    """
+    Class that inherits from both `torch.Tensor` and `MetaObj`, adding support for meta
+    data.
+
+    Metadata is stored in the form of a dictionary. Nested, an affine matrix will be
+    stored. This should be in the form of `torch.Tensor`.
+
+    Behavior should be the same as `torch.Tensor` aside from the extended
+    meta functionality.
+
+    Copying of information:
+        * For `c = a + b`, then auxiliary data (e.g., metadata) will be copied from the
+        first instance of `MetaTensor`.
+
+    Example:
+        .. code-block:: python
+
+            import torch
+            from monai.data import MetaTensor
+
+            t = torch.tensor([1,2,3])
+            affine = torch.eye(4)
+            meta = {"some": "info"}
+            m = MetaTensor(t, affine=affine, meta=meta)
+            m2 = m+m
+            assert isinstance(m2, MetaTensor)
+            assert m2.meta == meta
```

Comment:
Strictly speaking m2.meta is not equal to meta as m2.meta will have "affine" key in addition.

Reply:
good point, thanks!